linera-execution/src/resources.rs

```diff
@@ -268,6 +317,91 @@ where
         self.update_balance(self.policy.fuel_price(fuel, vm_runtime)?)
     }
 
+    /// Tracks runtime reading of `ChainId`
+    pub(crate) fn track_runtime_chain_id(&mut self) -> Result<(), ExecutionError> {
+        self.track_size_runtime_operations(RUNTIME_CHAIN_ID_SIZE)
+    }
+
+    /// Tracks runtime reading of `BlockHeight`
+    pub(crate) fn track_runtime_block_height(&mut self) -> Result<(), ExecutionError> {
+        self.track_size_runtime_operations(RUNTIME_BLOCK_HEIGHT_SIZE)
+    }
+
+    /// Tracks runtime reading of `ApplicationId`
+    pub(crate) fn track_runtime_application_id(&mut self) -> Result<(), ExecutionError> {
+        self.track_size_runtime_operations(RUNTIME_APPLICATION_ID_SIZE)
+    }
+
+    /// Tracks runtime reading of application parameters.
+    pub(crate) fn track_runtime_application_parameters(
+        &mut self,
+        parameters: &[u8],
+    ) -> Result<(), ExecutionError> {
+        let parameters_len = parameters.len() as u32;
+        self.track_size_runtime_operations(parameters_len)
+    }
+
+    /// Tracks runtime reading of `Timestamp`
+    pub(crate) fn track_runtime_timestamp(&mut self) -> Result<(), ExecutionError> {
+        self.track_size_runtime_operations(RUNTIME_TIMESTAMP_SIZE)
+    }
+
+    /// Tracks runtime reading of balance
+    pub(crate) fn track_runtime_balance(&mut self) -> Result<(), ExecutionError> {
+        self.track_size_runtime_operations(RUNTIME_AMOUNT_SIZE)
+    }
+
+    /// Tracks runtime reading of owner balances
+    pub(crate) fn track_runtime_owner_balances(
+        &mut self,
+        owner_balances: &[(AccountOwner, Amount)],
+    ) -> Result<(), ExecutionError> {
+        let mut size = 0;
+        for (account_owner, _) in owner_balances {
+            size += account_owner.size() + RUNTIME_AMOUNT_SIZE;
+        }
+        self.track_size_runtime_operations(size)
+    }
+
+    /// Tracks runtime reading of owners
+    pub(crate) fn track_runtime_owners(
+        &mut self,
+        owners: &[AccountOwner],
+    ) -> Result<(), ExecutionError> {
+        let mut size = 0;
+        for owner in owners {
+            size += owner.size();
+        }
+        self.track_size_runtime_operations(size)
+    }
+
+    /// Tracks runtime reading of owners
+    pub(crate) fn track_runtime_chain_ownership(
+        &mut self,
+        chain_ownership: &ChainOwnership,
+    ) -> Result<(), ExecutionError> {
+        let mut size = 0;
+        for account_owner in &chain_ownership.super_owners {
+            size += account_owner.size();
+        }
+        for account_owner in chain_ownership.owners.keys() {
+            size += account_owner.size() + RUNTIME_BLOCK_HEIGHT_SIZE;
```

Comment:
Maybe `RUNTIME_OWNER_WEIGHT_SIZE` should be a separate constant, even if it's the same value.

Reply:
Good catch!
I thought it was some block height.